.github/workflows/build_and_run_selected_test.yml

```diff
@@ -0,0 +1,117 @@
+name: Build and Run Selected Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      test-selection:
+        description: 'Specification of selected test(s), as accepted by pytest -k'
+        required: true
+        type: string
+      run-count:
+        description: 'Number of test runs to perform'
+        required: true
+        type: number
+      archs:
+        description: 'Archs to run tests on, e. g.: x64, arm64'
+        default: 'x64, arm64'
+        required: true
+        type: string
+      build_types:
+        description: 'Build types to run tests on, e. g.: debug, release'
+        default: 'debug'
+        required: true
+        type: string
+      pg-version:
+        description: 'Postgres version to use for testing (e. g. v17)'
+        default: 'v17'
+        required: true
+        type: string
+  pull_request:
+
+defaults:
+  run:
+    shell: bash -euxo pipefail {0}
+
+env:
+  RUST_BACKTRACE: 1
+  COPT: '-Werror'
+
+jobs:
+  meta:
+    uses: ./.github/workflows/_meta.yml
+    with:
+      github-event-name: ${{ github.event_name }}
+
+  build-and-test-locally:
+    needs: [ meta ]
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: ${{ fromJson(format('[{0}]', inputs.archs)) }}
+        build-type: ${{ fromJson(format('[{0}]', inputs.build_types)) }}
```

Comment:
I don't think this will create a correct JSON, for example with 2 items it's going to be `'[x64, arm64]'`

Reply:
Yeah, I'm yet to debug it, it was a blind change.

Reply:
Fixed.